src/lib.rs

```diff
@@ -0,0 +1,534 @@
+use std::sync::{Arc, Mutex};
+use wai_bindgen_rust::Handle;
+
+wai_bindgen_rust::export!("rustfft.wai");
+
+struct Rustfft {}
+impl rustfft::Rustfft for Rustfft {}
+pub struct Complex(original::num_complex::Complex64);
+
+impl Complex {
+    pub fn new(state: original::num_complex::Complex64) -> Self {
+        Self(state)
+    }
+}
+
+impl rustfft::Complex for Complex {
+    fn new_f64(re: f64, im: f64) -> Handle<Complex> {
+        Handle::new(Complex::new(original::num_complex::Complex64::new(re, im)))
+    }
+
+    fn i(&self) -> Handle<Complex> {
+        Handle::new(Complex::new(original::num_complex::Complex64::i()))
+    }
+}
+
+impl From<Complex> for original::num_complex::Complex64 {
+    fn from(complex: Complex) -> Self {
+        Self {
+            re: complex.0.re,
+            im: complex.0.im,
+        }
+    }
+}
+
+pub struct InnerFft(Arc<dyn original::Fft<f64> + 'static>);
```

Comment:
You don't need this `+ 'static` because it is implied.

```suggestion
pub struct InnerFft(Arc<dyn original::Fft<f64>>);
```

Reply:
removed `'static`